src/YourEasyRent/Services/State/UserStateManager.cs

```diff
@@ -0,0 +1,78 @@
+﻿namespace YourEasyRent.Services.State
+{
+    public class UserStateManager : IUserStateManager
+    {
+        public bool IsCategoryChosen { get; private set; }
+        public bool IsBrandChosen { get; private set; }
```

Comment:
You dont need ot explicitly set  `IsBrandChosen = true;`
You can use the list `menuStatuses.Any(x=>x =="BrandChosen");`